lib/vrl/stdlib/src/compact.rs

```diff
@@ -183,19 +183,17 @@ impl Expression for CompactFn {
             Value::Array(arr) => Ok(Value::from(compact_array(arr, &options))),
             value => Err(value::Error::Expected {
                 got: value.kind(),
-                expected: Kind::Array | Kind::Object,
+                expected: Kind::array(BTreeMap::default()) | Kind::object(BTreeMap::default()),
```

Comment:
I've seen this in a couple of places, and think the intention could made be clearer using

```suggestion
                expected: Kind::array(Collection::any()) | Kind::object(Collection::any()),
```

Reply:
Agreed, I started using `Collection::any()` at some point, but forgot to backport that to these cases, will update.

Reply:
Resolved in eead5d5.